Sources/Citadel/OpenSSHKey.swift

```diff
@@ -0,0 +1,223 @@
+import CCryptoBoringSSL
+import BigInt
+import Foundation
+import NIO
+import Crypto
+import NIOSSH
+
+internal protocol ReadableFromBuffer {
```

Comment:
I get why you wanted to discuss the name of this protocol, but since it's marked `internal` I can't complain. The only thing I can think of is changing it to `ParsableFromByteBuffer`, making it more explicit that this should _consume_ the ByteBuffer, and not any other buffer either.